Algorithm.CSharp/CoinbaseCryptoYearMarketTradingRegressionAlgorithm.cs

```diff
@@ -0,0 +1,141 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data;
+using QuantConnect.Brokerages;
+using QuantConnect.Interfaces;
+using System.Collections.Generic;
+using QuantConnect.Securities.Crypto;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    public class CoinbaseCryptoYearMarketTradingRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        /// <summary>
+        /// The Average amount of day in year
+        /// </summary>
+        /// <remarks>Regardless of calendar</remarks>
+        private const int _daysInYear = 365;
+
+        /// <summary>
+        /// Flag prevents same order <see cref="Orders.OrderDirection"/>
+        /// </summary>
+        private bool _isBuy;
+
+        /// <summary>
+        /// Trading security
+        /// </summary>
+        private Crypto _BTCUSD;
+
+        /// <summary>
+        /// Initialize the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        /// <remarks>In fact, you can assign custom value for <see cref="IAlgorithmSettings.TradingDaysPerYear"/></remarks>
+        public override void Initialize()
+        {
+            SetStartDate(2015, 01, 13);
+            SetEndDate(2016, 02, 03);
+
+            SetCash(100000);
+
+            // Setup brokerage for current algorithm
+            SetBrokerageModel(BrokerageName.Coinbase, AccountType.Cash);
+
+            _BTCUSD = AddCrypto("BTCUSD", Resolution.Daily, Market.Coinbase);
+        }
+
+        /// <summary>
+        /// Data Event Handler: receiving all subscription data in a single event
+        /// </summary>
+        /// <param name="slice">The current slice of data keyed by symbol string</param>
+        public override void OnData(Slice slice)
+        {
+            if (!_isBuy)
+            {
+                MarketOrder(_BTCUSD, 1);
+                _isBuy = true;
+            }
+            else
+            {
+                Liquidate();
+                _isBuy = false;
+            }
+        }
```

Comment:
This is not really necessary, we just need to assert the trading days per year value